crates/pathfinder/src/rpc/api.rs

```diff
@@ -99,9 +101,10 @@ impl RpcApi {
                     .await
                     .map_err(internal_server_error)?;
 
+                let old_root = self.get_latest_block_root().await?;
```

Comment:
How does this work if we haven't yet synced up to the latest block?

In this case I guess we'd just fetch latest state root from our DB and return the pending block with old_root == root state from our latest, which might be incorrect?

Reply:
You're right, I need to have a second look.

Reply:
I'm moving this to a separate PR.